tmt/utils.py

```diff
@@ -2102,3 +2102,31 @@ def update(self, value: str, color: Optional[str] = None) -> None:
 
         sys.stdout.write(f"\r{message}")
         sys.stdout.flush()
+
+
+def find_fmf_root(path: str, just_top: bool = False) -> List[str]:
+    """
+    Search trough path and return all fmf roots that exist there
+
+    If `just_top` is True it will return single item
+
+    Raise MetadataError if no fmf root is found
+    """
+    min_len = None
+    top_path = ''
+    fmf_roots = []
+    for root, _, files in os.walk(path):
+        if not os.path.basename(root) == '.fmf':
+            continue
+        if 'version' in files:
+            candidate = os.path.dirname(root)
+            fmf_roots.append(candidate)
+            if just_top and (min_len is None or len(candidate) < min_len):
```

Comment:
So for the `top_path` we are choosing the directory with the shortest name? What's the motivation for this choise?

Reply:
E.g. tmt repo contains several fmf roots: main one is in git root and many in tests inside 'data' subdirectories. 
Motivation is to find one in git root and ignore those in data subdirectories - the git one has shortest path ...
And it should work also if fmf root is one level down, eg. for following structure:
```
./tests/super/test/data/.fmf/version
./tests/.fmf/version
./source
```

Reply:
I see. Thanks for explanation. So this works when fmf trees are nested and returns the shortest directory when multiple trees are on the same level, e.g.:

    >>> tmt.utils.find_fmf_root("examples")
    ['examples/httpd', 'examples/mini', 'examples/systemd', 'examples/ansible', 'examples/convert', 'examples/discover', 'examples/environment', 'examples/inherit', 'examples/l2', 'examples/local', 'examples/manual', 'examples/multiple', 'examples/rtt', 'examples/symlinks', 'examples/together', 'examples/wow']

    >>> tmt.utils.find_fmf_root("examples", just_top=True)
    ['examples/l2']

Sounds ok.